gax-java/gax/src/test/java/com/google/api/gax/retrying/FailingCallable.java

```diff
@@ -66,20 +67,29 @@ class FailingCallable implements Callable<String> {
   private final String result;
   private final CountDownLatch firstAttemptFinished = new CountDownLatch(1);
 
+  private volatile RetryingFuture<String> externalFuture;
+
   FailingCallable(int expectedFailuresCount, String request, String result, ApiTracer tracer) {
     this.request = request;
     this.tracer = tracer;
     this.expectedFailuresCount = expectedFailuresCount;
     this.result = result;
   }
 
+  public void setExternalFuture(RetryingFuture<String> externalFuture) {
+    this.externalFuture = Preconditions.checkNotNull(externalFuture);
+  }
+
   CountDownLatch getFirstAttemptFinishedLatch() {
     return firstAttemptFinished;
   }
 
   @Override
   public String call() throws Exception {
     try {
+      if (externalFuture.isDone()) {
```

Comment:
```suggestion
      if (externalFuture != null && externalFuture.isDone()) {
```
We can add a null check so that we don't have to set it for existing tests.

Reply:
Yeah that is an option too. I was thinking that all existing and future tests *should* pass in the external future (which more accurately mimics the AttemptCallable) so it shouldn't be null. WDYT?

Reply:
I see, yes that makes sense, can you add some comments to reflect the idea?

Reply:
Sure, will do.